spec/acceptance/03_virtualbox_type_spec.rb

```diff
@@ -0,0 +1,355 @@
+require 'spec_helper_acceptance'
+
+describe 'virtualbox type/provider' do
+  context 'creating a complex VM' do
+    it 'should be idempotent after 2 runs' do
+      pp = <<-EOS
+      virtual_machine { 'computer':
+        ensure          => present,
+        ostype          => 'Other',
+        register        => true,
+        state           => 'running',
+        description     => 'Managed by puppet',
+        memory          => 512,
+        pagefusion      => 'off',
+        vram            => 16,
+        acpi            => 'on',
+        nestedpaging    => 'on',
+        largepages      => 'on',
+        longmode        => 'off',
+        synthcpu        => 'off',
+        hardwareuuid    => '3e039b44-0b67-4109-8936-f9e3ac519c9b',
+        cpus            => 1,
+        cpuexecutioncap => 100,
+        monitorcount    => 1,
+        accelerate3d    => 'off',
+        firmware        => 'BIOS',
+        boot1           => 'net',
+        boot2           => 'dvd',
+        boot3           => 'disk',
+        boot4           => 'none',
+        io_apic         => 'on',
+        nics            => {
+          1 => {
+            mode  => 'nat',
+            type  => 'Am79C973',
+            speed => 0
+          }
+        }
+      }
+      EOS
+
+      # We are applying the manifest twice here due to the fact that you have
+      # to apply a manifest twice when creating a VM, I will look for a
+      # workaround fo this at a later date.
+      apply_manifest(pp, :catch_failures => true)
+      apply_manifest(pp, :catch_failures => true)
+      apply_manifest(pp, :catch_changes => true)
+    end
+
+    it 'should have been created' do
+      command('VBoxManage showvminfo computer') do
+        its(:exit_status) { should eq 0 }
+        its(:exit_status) { should eq 0 }
+        its(:stdout) { should match /Name:\s*computer/ }
+      end
+    end
+
+    it 'should be running' do
+      command('VBoxManage list runningvms') do
+        its(:exit_status) { should eq 0 }
+        its(:stdout) { should match /computer/ }
+      end
+    end
+
+    it 'should have the correct description' do
+      command('VBoxManage showvminfo computer --machinereadable') do
+        its(:exit_status) { should eq 0 }
+        its(:stdout) { should match /description="Managed by puppet"/ }
+      end
+    end
+
```

Comment:
Also, I think these 'should have the correct *' could be compressed into one example:

```ruby
it 'should have the correct settings' do
  command('VBoxManage showvminfo computer --machinereadable') do
    its(:exit_status) { should eq 0 }
    its(:stdout) do
      should match /description="Managed by puppet"/
      should match /type=Other\/Unknown/
      should match /memory=512/
      ...and so on...
    end
  end
end

This way we only have to execute `VBoxManage showvminfo computer --machinereadable` once!